ui/gtk3/emojier.vala

```diff
@@ -1239,9 +1239,47 @@ public class IBusEmojier : Gtk.ApplicationWindow {
     private static GLib.SList<string>?
     lookup_emojis_from_annotation(string annotation) {
         GLib.SList<string>? total_emojis = null;
+        GLib.SList<string>? non_glyph_emojis = null;
         unowned GLib.SList<string>? sub_emojis = null;
         unowned GLib.SList<unichar>? sub_exact_unicodes = null;
         unowned GLib.SList<unichar>? sub_unicodes = null;
+        // Create Pango context for font checking
+        var label = new Gtk.Label("");
+        var pango_context = label.get_pango_context();
+        var layout = new Pango.Layout(pango_context);
+        var font_desc = Pango.FontDescription.from_string(m_emoji_font_family);
+        layout.set_font_description(font_desc);
+        bool is_emoji_supported(string emoji) {
```

Comment:
Please rename `is_emoji_supported()` to `is_visible()` and move the inner function to an API of the new class likes `emojier.ECheckVisibleLabel.is_visible()`.